File: wf-api-integration/references/statement-inquiry/inquiry-statement-list/python/model/response/inquiry_statement_response.py
# -*- coding: utf-8 -*-
"""
WorldFirst inquiryStatementList 响应模型

@author Qoder
@version inquiry_statement_response.py, v 0.1 2026-04-14
"""

from dataclasses import dataclass, field
from typing import List, Optional

from {basePackage}.wf.model.response.result import Result
from {basePackage}.wf.model.domain.statement_record import StatementRecord


@dataclass
class InquiryStatementResponse:
    """inquiryStatementList 响应

    Attributes:
        result: 通用响应结果
        response_id: 响应唯一 ID
        fee_item_type: 手续费类型
        statement_list: 流水记录列表
        total_count: 总记录数
        total_page_number: 总页数
        current_page_number: 当前页码
    """

    result: Optional[Result] = None
    response_id: Optional[str] = None
    fee_item_type: Optional[str] = None
    statement_list: Optional[List[StatementRecord]] = None
    total_count: Optional[int] = None
    total_page_number: Optional[int] = None
    current_page_number: Optional[int] = None

    @classmethod
    def from_dict(cls, data: dict) -> "InquiryStatementResponse":
        """从字典创建响应对象

        Args:
            data: 响应字典

        Returns:
            InquiryStatementResponse 实例
        """
        if data is None:
            return cls()

        statement_list_data = data.get("statementList")
        statement_list = None
        if statement_list_data is not None:
            statement_list = [StatementRecord.from_dict(item) for item in statement_list_data]

        return cls(
            result=Result.from_dict(data.get("result")),
            response_id=data.get("responseId"),
            fee_item_type=data.get("feeItemType"),
            statement_list=statement_list,
            total_count=data.get("totalCount"),
            total_page_number=data.get("totalPageNumber"),
            current_page_number=data.get("currentPageNumber"),
        )

File: wf-api-integration/references/statement-management/inquiry-statement-list/java/model/response/InquiryStatementResponse.java
package {basePackage}.wf.model.response;

import org.apache.commons.lang3.builder.ToStringBuilder;
import org.apache.commons.lang3.builder.ToStringStyle;
import {basePackage}.wf.model.domain.StatementRecord;

import java.util.List;

/**
 * WorldFirst inquiryStatementList 响应对象
 *
 */
public class InquiryStatementResponse {

    /** 接口调用结果 */
    private Result result;

    /** 响应唯一 ID，最大 32 位 */
    private String responseId;

    /** 费项类型：OBO_SERVICE_FEE、REMIT_SERVICE_FEE */
    private String feeItemType;

    /** 账单流水列表 */
    private List<StatementRecord> statementList;

    /** 总记录数 */
    private Integer totalCount;

    /** 总页数 */
    private Integer totalPageNumber;

    /** 当前页码 */
    private Integer currentPageNumber;

    /**
     * Getter method for property <tt>result</tt>.
     *
     * @return property value of result
     */
    public Result getResult() {
        return result;
    }

    /**
     * Setter method for property <tt>result</tt>.
     *
     * @param result value to be assigned to property result
     */
    public void setResult(Result result) {
        this.result = result;
    }

    /**
     * Getter method for property <tt>responseId</tt>.
     *
     * @return property value of responseId
     */
    public String getResponseId() {
        return responseId;
    }

    /**
     * Setter method for property <tt>responseId</tt>.
     *
     * @param responseId value to be assigned to property responseId
     */
    public void setResponseId(String responseId) {
        this.responseId = responseId;
    }

    /**
     * Getter method for property <tt>feeItemType</tt>.
     *
     * @return property value of feeItemType
     */
    public String getFeeItemType() {
        return feeItemType;
    }

    /**
     * Setter method for property <tt>feeItemType</tt>.
     *
     * @param feeItemType value to be assigned to property feeItemType
     */
    public void setFeeItemType(String feeItemType) {
        this.feeItemType = feeItemType;
    }

    /**
     * Getter method for property <tt>statementList</tt>.
     *
     * @return property value of statementList
     */
    public List<StatementRecord> getStatementList() {
        return statementList;
    }

    /**
     * Setter method for property <tt>statementList</tt>.
     *
     * @param statementList value to be assigned to property statementList
     */
    public void setStatementList(List<StatementRecord> statementList) {
        this.statementList = statementList;
    }

    /**
     * Getter method for property <tt>totalCount</tt>.
     *
     * @return property value of totalCount
     */
    public Integer getTotalCount() {
        return totalCount;
    }

    /**
     * Setter method for property <tt>totalCount</tt>.
     *
     * @param totalCount value to be assigned to property totalCount
     */
    public void setTotalCount(Integer totalCount) {
        this.totalCount = totalCount;
    }

    /**
     * Getter method for property <tt>totalPageNumber</tt>.
     *
     * @return property value of totalPageNumber
     */
    public Integer getTotalPageNumber() {
        return totalPageNumber;
    }

    /**
     * Setter method for property <tt>totalPageNumber</tt>.
     *
     * @param totalPageNumber value to be assigned to property totalPageNumber
     */
    public void setTotalPageNumber(Integer totalPageNumber) {
        this.totalPageNumber = totalPageNumber;
    }

    /**
     * Getter method for property <tt>currentPageNumber</tt>.
     *
     * @return property value of currentPageNumber
     */
    public Integer getCurrentPageNumber() {
        return currentPageNumber;
    }

    /**
     * Setter method for property <tt>currentPageNumber</tt>.
     *
     * @param currentPageNumber value to be assigned to property currentPageNumber
     */
    public void setCurrentPageNumber(Integer currentPageNumber) {
        this.currentPageNumber = currentPageNumber;
    }

    @Override
    public String toString() {
        return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE);
    }
    }
}
